server/channels/api4/user.go

```diff
@@ -2270,6 +2272,38 @@ func attachDeviceId(c *Context, w http.ResponseWriter, r *http.Request) {
 	ReturnStatusOK(w)
 }
 
+func setExtraSessionProps(c *Context, w http.ResponseWriter, r *http.Request) {
```

Comment:
Does every device send this request to the server when they start a new session? Or only the ones that we know for sure can't receive notifications?

Reason I ask is that I think it would actually make more sense to have the mobile devices tell the app that they WILL be able to show notifications rather than they won't, to cover other edge cases. But I can see that being a lot if we expect the API traffic to be too heavy.

Reply:
The idea is that this will be sent on each startup, saying either true or false, and the version number.

So... yeah, it will be more traffic, but hopefully nothing too bad. That is the reason why I refrain from going down to the database if the values are the same, since most of the times, the values won't change.

Reply:
If we are planning on doing that for every device, then I would suggest changing the logic from sending a flag to disable the ACK tracking, in favour of one that enables it.

This ensures:
1) We use the same logic as the webapp, the webapp also doesn't expects ACKs unless explicitly told by the client
2) It means that any devices that fail to send this information we won't expect ACKs from, which is probably preferable to expecting ones to disable it, hopefully leading to less false positives wrt. missing notifications.

Reply:
After our talk and thinking about it a bit more, there is still one small problem. "Old app versions". In general we don't support them, but anything that is before the change on the mobile side will be not registering acks. This is because those clients will not be sending the information about whether they have notifications enabled or not.

Since most people have their apps up to date, we don't really support old app versions, and we are only dropping the ack, not the whole notification, it should be fine, but we have to keep that in mind.

@marianunez do you foresee any other issue? Or are you fine with this small change?

Reply:
> If we are planning on doing that for every device, then I would suggest changing the logic from sending a flag to disable the ACK tracking, in favour of one that enables it.

Would it enable it at start up and the notifications are enabled? Not sure I'm following where we only disable the ACK

Reply:
> Right now, it is "default ack". So... if the device has said nothing, ACK the notifications.

I agree we having the default be `ACK` which is the current behavior without this PR.  Agreed that having to support older devices we will lose even more ack.